spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/CloudSqlEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,203 @@
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.util.ClassUtils;
+
+/**
+ * Provides Google Cloud SQL instance connectivity through Spring JDBC by providing only a
+ * database and instance connection name.
+ *
+ * @author João André Martins
+ * @author Artem Bilan
+ * @author Mike Eltsufin
+ * @author Chengyuan Zhao
+ * @author Eddú Meléndez
+ */
+public class CloudSqlEnvironmentPostProcessor implements EnvironmentPostProcessor {
+	private final static String CLOUD_SQL_PROPERTIES_PREFIX = "spring.cloud.gcp.sql.";
+
+	private static final Log LOGGER =
+			LogFactory.getLog(CloudSqlEnvironmentPostProcessor.class);
+
+	@Override
+	public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+		DatabaseType databaseType = getEnabledDatabaseType(environment);
+
+		if (databaseType != null) {
+			if (LOGGER.isInfoEnabled()) {
+				LOGGER.info("post-processing Cloud SQL properties for + " + databaseType.name());
+			}
+
+			CloudSqlJdbcInfoProvider cloudSqlJdbcInfoProvider = buildCloudSqlJdbcInfoProvider(environment, databaseType);
+
+			// configure default JDBC driver and username as fallback values when not specified
+			Map<String, Object> fallbackMap = new HashMap<>();
+			fallbackMap.put("spring.datasource.username", "root");
+			fallbackMap.put("spring.datasource.driver-class-name", cloudSqlJdbcInfoProvider.getJdbcDriverClass());
+			environment.getPropertySources()
+					.addLast(new MapPropertySource("CLOUD_SQL_DATA_SOURCE_FALLBACK", fallbackMap));
+
+			// always set the spring.datasource.url property in the environment
+			Map<String, Object> primaryMap = new HashMap<>();
+			primaryMap.put("spring.datasource.url", cloudSqlJdbcInfoProvider.getJdbcUrl());
+			environment.getPropertySources()
+					.addFirst(new MapPropertySource("CLOUD_SQL_DATA_SOURCE_URL", primaryMap));
+
+			setCredentials(environment, application);
+
+			// support usage metrics
+			CoreSocketFactory.setApplicationName("spring-cloud-gcp-sql/"
+					+ this.getClass().getPackage().getImplementationVersion());
+		}
+	}
+
+	private DatabaseType getEnabledDatabaseType(ConfigurableEnvironment environment) {
+		if (Boolean.parseBoolean(getSqlProperty(environment, "enabled", "true"))
+				&& isOnClasspath("javax.sql.DataSource")
+				&& isOnClasspath("org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType")
+				&& isOnClasspath("com.google.cloud.sql.CredentialFactory")) {
+			if (isOnClasspath("com.google.cloud.sql.mysql.SocketFactory")
+					&& isOnClasspath("com.mysql.cj.jdbc.Driver")) {
+				return DatabaseType.MYSQL;
+			}
+			else if (isOnClasspath("com.google.cloud.sql.postgres.SocketFactory")
+					&& isOnClasspath("org.postgresql.Driver")) {
+				return DatabaseType.POSTGRESQL;
+			}
+		}
+		return null;
+	}
+
+	private boolean isOnClasspath(String className) {
```

Comment:
Is there no Spring utility we can use for this?

Reply:
I didn't find it. They do exactly the same thing in their `@ConditionalOnClass` implementation, but it's not reusable.

Reply:
I don't know what version you looked at but `OnClassCondition` uses `ClassUtils#isPresent` that does what this method does.

Reply:
I might've actually seen it somewhere else. `ClassUtils#isPresent` works, Thanks!